### DISCUSSION TITLE
Docker image template field

### DISCUSSION THREAD

**Discussion by calvinbui** [renovatebot/renovate#17313] [Request Help] [answered]
I'm using Renovate for updating Docker image version inside my Ansible playbooks.

For example. this PR: [ajanvier/polr: update version to v2.3.0](https://github.com/calvinbui/ansible-monorepo/pull/29)

The `{{depName}}` field gives me `ajanvier/polr`, but is there a way to just get the image name, i.e. `polr`?

I've considered:

1. Handlebars do not support inline formatting.
2. Regex Manager, but the docs don't seem to suit it.
3. Going through the [template fields docs](https://docs.renovatebot.com/templates/)

**Comment by MaronHatoum** [comment, marked as answer]
Hi @calvinbui 
you could try using the following `regexManager` :

```
{
  "regexManagers": [
    {
      "fileMatch": ["(^|\/|\\.)(polr\.yaml)$"],
      "matchStrings": ["\s*image:\s+?(?<orgRepo>[a-z]+)\/(?<depName>[\/a-z]+)(?::(?<currentValue>[a-z0-9.-]+))?"],
    }
  ]
}
```
**regexManagers docs:** https://docs.renovatebot.com/modules/manager/regex/

**Reply by calvinbui** [reply]
Thanks, that regex looks good, but there are some cases it doesn't match, e.g.,:

- `image: ghcr.io/home-assistant/home-assistant:2022.8.6`
- `image: eclipse-mosquitto:2.0.15`

**Reply by MaronHatoum** [reply]
I've updated the regex.

```
{
  "regexManagers": [
    {
      "fileMatch": ["(^|\/|\\.)(polr\.yaml)$"],
      "matchStrings": ["\s*image:\s+?(:?[a-z]+\/)*(?<depName>[\/a-z]+)(?::(?<currentValue>[a-z0-9.-]+))?"],
    }
  ]
}
```

**Reply by calvinbui** [reply]
I've made some changes to your regex and tested on regex101.com. I'll apply this and give it a try tomorrow.

```
\s*image:\s+?(:?[\/a-z\.-]+\/)*(?<depName>[\/a-z-]+)(?::(?<currentValue>[a-z0-9.-]+))?
```

![image](https://user-images.githubusercontent.com/3604363/185931732-4cda860a-dd78-44e4-b343-4cc3319b22ed.png)

**Reply by viceice** [reply]
you probably also need to match the full name as `packageName`, so renovate knows the full name for update lookup

**Reply by calvinbui** [reply]
thanks guys, that did it, here's the final config

```
  regexManagers: [
    // extract depName from Docker image
    {
      datasourceTemplate: "docker",
      fileMatch: ["^.*\\.ya?ml$"],
      matchStrings: ["\\s*image:\\s+?(?<registry>:?[\\/a-z\\.-]+\\/)*(?<depName>[\\/a-z-]+)(?::(?<currentValue>[a-z0-9.-]+))?"],
      packageNameTemplate: "{{{ registry }}}{{{ depName }}}"
    },
  ],
```

Before:

![image](https://user-images.githubusercontent.com/3604363/185939516-cfd6621d-e633-4d62-b95b-b6fe5123aaa4.png)

After:

![image](https://user-images.githubusercontent.com/3604363/185939408-548201a7-a766-4ee0-bc45-c8f3406edbb1.png)
